Scripts/BulkConversion/SharedScriptLib.ps1

```diff
@@ -0,0 +1,91 @@
+    [Parameter(Position=2)] [int]     $Severity  = 1,
+    [Parameter(Position=3)] [boolean] $WriteHost = $true,
+                            [string]  $Path      = $("C:\Temp\Log"),
+                            [string]  $textcolor = "White"
+)
+    IF(!(Test-path -Path $Path)) {$Scratch = mkdir $Path}
+    $Error.Clear()
+
+    #Formats the values required to enter for Trace32 Format
+    #$TimeZoneBias = Get-WmiObject -Query "Select Bias from Win32_TimeZone"
+    [string]$Time = Get-Date -Format "HH:mm:ss.ffff"
+    [string]$Date = Get-Date -Format "MM-dd-yyyy"
+
+    #Appends the newest log entry to the end of the log file in a Trace32 Formatting
+    $('<![LOG['+$LogValue+']LOG]!><time="'+$Time+'" date="'+$Date+'" component="'+$component+'" context="Empty" type="'+$severity+'" thread="Empty" file="'+"Empty"+'">') | out-file -FilePath $($Path+"\BulkConversion.log") -Append -NoClobber -encoding default -ErrorAction SilentlyContinue -ErrorVariable LogError
+
+    ## If Writing to log file fails try again.
+    While ($Error.count -gt 0)
+    {
+        ## Gives a random amount of time to wait until next write attempt.
+        $Error.Clear()
+        Sleep($(get-random -Maximum 0.5 -Minimum 0.0))
+
+        #$('<![LOG['+$LogError+']LOG]!><time="'+$Time+'" date="'+$Date+'" component="'+$component+'" context="Empty" type="'+3+'" thread="Empty" file="'+"Empty"+'">') | out-file -FilePath $($Path+"\BulkConversion.log") -Append -NoClobber -encoding default -ErrorAction SilentlyContinue -ErrorVariable LogError
+
+        #Appends the newest log entry to the end of the log file in a Trace32 Formatting
+        $('<![LOG['+$LogValue+']LOG]!><time="'+$Time+'" date="'+$Date+'" component="'+$component+'" context="Empty" type="'+$severity+'" thread="Empty" file="'+"Empty"+'">') | out-file -FilePath $($Path+"\BulkConversion.log") -Append -NoClobber -encoding default -ErrorAction SilentlyContinue -ErrorVariable LogError
+    }
+
+    $LogValue | Out-File -Append -FilePath "$Path\BulkConversion-Log.txt"
+    IF($WriteHost)
+    {
+        $LogValue | Out-File -Append -FilePath "$Path\BulkConversion-Display.txt"
+        Write-Host $("" + $LogValue) -ForegroundColor $(switch ($Severity) {1 {$textcolor} 2 {"Yellow"} 3 {"Red"}})
+    }
+}
+
+Function New-InitialSnapshot ($SnapshotName, $VMName, $jobId="-" )
+{
+    $FunctionName = Get-FunctionName
+    ## Verifies if the script snapshot exists, if not exists snapshot is created.
+    IF ($SnapshotName -cnotin $(Get-VMSnapshot -VMName $vmName).Name)
+    {
+        New-LogEntry -LogValue "Creating VM Snap for VM ($VMName): $SnapshotName" -Component "JobID($JobID) - $FunctionName" 
+        $Scratch = Checkpoint-VM -Name $vmName -SnapshotName "$SnapshotName"
+    }
+    Else
+    {
+        New-LogEntry -LogValue "Snapshot ($SnapshotName) for VM ($VMName) already exists. " -Component "JobID($JobID) - $FunctionName"
+    }
+}
+
+Function Restore-InitialSnapshot ($SnapshotName, $VMName, $jobId="" )
+{
```

Comment:
validate params

Reply:
Updated the Function Parameters to include value validation.